Multiple Bug Fixes and Refactors:

paolopaolopaolo opened the pull request:
- Fixes #26: Add `MEDIA_ROOT` and `MEDIA_URL`
- Fixes #28: Fix Vagrantfile synced folder src and dest
- Fixes #30: Add step in orchestration that fixes virtualenvs permissions
- Fixes #27: Remove dependency installation and setup from flow (and add to readme)
- Remove virtualenvwrapper requirement, add ansible requirement on README
- Fix weirdness happening on `setup.existing`

@rmutter, @baylee: Bunch of fixes to some broken parts. Big thing is that I thought hard about why we'd be doing things with virtualenvwrapper in the first place (on the local, non-virtual machine). I wasn't sold on making users setup virtualenvwrapper and then setup a specific virtualenv with virtualenvwrapper, so we'll now expect users to have ansible installed in their local environment.

paolopaolopaolo commented on fabfile/requirements.txt:
```diff
@@ -1,26 +1,3 @@
-contextlib2==0.4.0
```
As far as I can tell, literally nothing. Potentially we can remove what's in there now and make it the requirements.txt file for Yurt dependencies, so that users don't have to `pip install` all the pre-reqs one at a time.

rmutter commented on django_project/config/settings/base.py:
```diff
@@ -1,23 +1,14 @@
+from os.path import dirname, abspath
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+CONFIG_ROOT = dirname(dirname(abspath(__file__)))
```
is this different then `BASE_DIR` above it?

rmutter commented on django_project/config/settings/base.py:
```diff
@@ -89,9 +82,8 @@
+STATIC_ROOT = os.path.join(PROJECT_ROOT, STATIC_URL.strip("/"))
+
+MEDIA_URL = STATIC_URL + "media/"
+MEDIA_ROOT = os.path.join(PROJECT_ROOT, *MEDIA_URL.strip("/").split("/"))
```
if we're stripping all `/` is there any to split on?

paolopaolopaolo replied:
https://docs.python.org/3/library/stdtypes.html#str.strip

Essentially, strip only removes `/` one at a time from either end until it hits a character that isn't `/`. That means the `/` between "static" and "media" will be spared.

rmutter commented on fabfile/setup.py:
```diff
@@ -223,19 +190,17 @@ def existing():
-    with ansible():
-        local("vagrant plugin install vagrant-vbguest")
-        local("vagrant up")
+    local("vagrant up")
```
Would it be worth while checking somewhere that they have ansible installed and informing them to download it if it isn't? Basically I'm thinking a try/except around importing ansible. Now that we're not attempting to install some things for users, it may be worthwhile to give friendly error messages when they're not installed.

paolopaolopaolo replied:
We've been able to reduce the dependencies to just Fabric and Ansible (and their respective dependencies) and pycrypto-on-pypi for older/experimental versions of Python 2.7. I could make the root level requirements.txt file have these dependencies and just tell users to `pip install -r` them (so they don't miss anything).

paolopaolopaolo commented:
@rmutter, @baylee, simplified the flow in that in both existing and new projects, the fabfile runs `vagrant up`. I made a fleshed out Troubleshooting section in the README to handle issues that will crop up with existing projects setup in Ansible 1.9.4, and I simplified/reformatted the instructions in the README to be much easier to comprehend. I'm ready to merge this whenever you guys are.
